### ISSUE TITLE
Window Operations do not preserve ordering

### ISSUE THREAD

**Issue by hawkfish** [duckdblabs/duckplyr#41]
@krlmlr I now have a similar problem to #38 with `relational union() order-preserving` which you can see in https://github.com/duckdb/duckdb/pull/8593 Basically we now evaluate all window functions in parallel, so row number will not necessarily come out in row number order(!) In fact, if the data is not too large, it is likely that the last chunk will come out first as it will be faster.

**Comment by krlmlr** [comment]
This should be fixed in the main branch of this repo. If not, we can reopen or file a new issue.
